ta4j-core/src/main/java/org/ta4j/core/BaseBar.java

```diff
@@ -230,7 +232,11 @@ public void addTrade(Num tradeVolume, Num tradePrice) {
     }
 
     @Override
-    public void addPrice(Num price) {
+    public void addPrice(Num inPrice) {
+        Num price = inPrice;
+        if (numFunction != null) {
+            price = numFunction.apply(inPrice.getDelegate());
```

Comment:
The new instance variable is only needed for this statement? If `numFunction != null` and the user tries to add another `Num ` implementation the `inPrice` will be converted to the `Num ` implementation of the `BaseBar` class. Is this the correct behaviour? The user can get unexpected rounding deviatations. I think we should not implicit transform different Num implementations, but throw a `ClassCastException ` if the user tries to mix up different `Num ` implementations. There are also default functions in `Bar` interface that support adding and transforming a `Number` or `String` into the corresponding `Num` implementation of the `BaseBar` class:
```
    default void addPrice(String price, Function<Number, Num> numFunction){
        addPrice(numFunction.apply(new BigDecimal(price)));
    }

    default void addPrice(Number price, Function<Number, Num> numFunction){
        addPrice(numFunction.apply(price));
}
```

Reply:
If `numFunction` is not stored and used then what is the purpose of the `numFunction` parameter in the constructor?
```
public BaseBar(Duration timePeriod, ZonedDateTime endTime, Function<Number, Num> numFunction)
```

Reply:
You need the `numFunction` parameter to initialize some values:

```
this.volume = numFunction.apply(0);
this.amount = numFunction.apply(0);
```

Reply:
When I remove the change to `BaseBar` the following unit test in `PrecisionNumTest` (line 186) fails
```
assertEquals(32, ((BigDecimal) bigDecimalSeries.getBar(0).getClosePrice().getDelegate()).precision());
```
with
```
java.lang.AssertionError: expected:<32> but was:<120>
```